src/system/SystemLayer.h

```diff
@@ -288,6 +290,32 @@ class LayerSockets : public Layer
     virtual SocketWatchToken InvalidSocketWatchToken() = 0;
 };
 
+class LayerSocketsLoop;
+
+/**
+ * Enables the participation of subordinate event loops in the SystemLayer event loop.
+ */
+class EventLoopHandler : public chip::IntrusiveListNodeBase<>
+{
+public:
+    virtual ~EventLoopHandler() {}
+
+    /**
+     * Prepares events and returns the next requested wake time.
+     */
+    virtual Clock::Timestamp PrepareEvents(Clock::Timestamp now) { return Clock::Timestamp::max(); }
+
+    /**
+     * Handles / dispatches pending events.
+     * Every call to this method will have been preceded by a call to `PrepareEvents`.
+     */
+    virtual void HandleEvents() = 0;
+
+private:
+    friend class LayerSocketsLoop;
```

Comment:
why are you using friend classes? Friend classes create strong coupling and should be generally avoided. If we use friend, please explain in a comment. "Because others may mis-use it" is not a great reason (e.g. we probably want some form of unit tests which often want internal state access).

Reply:
It's explained in the comment one line below. It's so the event loop implementation can use `mState`. I can spell it out more explicitly.